pkg/session/cursor/state.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cursor
+
+// State is a read-only interface to access the state of cursor
+type State interface {
```

Comment:
Why we need to add a new interface `State` instead of just using a `uint64` or `struct` as the state

Reply:
I agree. The interface is over-complicated. I'll use a struct directly as instead.